plugins/modules/ndo_l3out_interface_routing_policy.py

```diff
@@ -0,0 +1,620 @@
+  cisco.mso.ndo_l3out_interface_routing_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    uuid: "{{ query_with_name.current.uuid }}"
+    state: query
+  register: query_with_uuid
+
+- name: Query all L3Out Interface Routing Policies
+  cisco.mso.ndo_l3out_interface_routing_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    state: query
+  register: query_all
+
+- name: Delete an L3Out Interface Routing Policy
+  cisco.mso.ndo_l3out_interface_routing_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    name: irp_1_updated
+    state: absent
+
+- name: Delete an L3Out Interface Routing Policy using UUID
+  cisco.mso.ndo_l3out_interface_routing_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_tenant_template
+    uuid: "{{ query_with_name.current.uuid }}"
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate
+from ansible_collections.cisco.mso.plugins.module_utils.constants import ENABLED_OR_DISABLED_TO_BOOL_STRING_MAP
+from ansible_collections.cisco.mso.plugins.module_utils.utils import append_update_ops_data
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        uuid=dict(type="str", aliases=["l3out_interface_routing_policy_uuid"]),
+        name=dict(type="str", aliases=["l3out_interface_routing_policy_name"]),
+        description=dict(type="str"),
+        bfd_settings=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                admin_state=dict(type="str", choices=["enabled", "disabled"]),
+                detection_multiplier=dict(type="int"),
+                min_receive_interval=dict(type="int"),
+                min_transmit_interval=dict(type="int"),
+                echo_receive_interval=dict(type="int"),
+                echo_admin_state=dict(type="str", choices=["enabled", "disabled"]),
+                interface_control=dict(type="str", choices=["enabled", "disabled"]),
+            ),
+        ),
+        bfd_multi_hop_settings=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                admin_state=dict(type="str", choices=["enabled", "disabled"]),
+                detection_multiplier=dict(type="int"),
+                min_receive_interval=dict(type="int"),
+                min_transmit_interval=dict(type="int"),
```

Comment:
this is similar to bfd_multi_hop_settings in ndo_l3out_node_routing_policy, so can we move this to generic place and use at both places?